api/backend/backend.go

```diff
@@ -52,6 +51,18 @@ var backends = struct {
 	r []*registeredBackend
 }{}
 
+var instance Service
+
+// Current return the active backend instance
+func Current() Service {
+	return instance
+}
+
+// WithBackend set the active backend instance
+func WithBackend(s Service) {
```

Comment:
For these singleton get methods, maybe we should have an explicit error saying this is not initialised, the corresponding set method should be called first, rather than let the code continue with uninitialised values